core/src/main/java/org/wildfly/channel/proxy/HttpUrlStreamHandler.java

```diff
@@ -0,0 +1,80 @@
+package org.wildfly.channel.proxy;
+
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.config.AuthSchemes;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHeader;
+
+import java.io.IOException;
+import java.net.Proxy;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.util.Collections;
+
+import static org.wildfly.channel.proxy.HttpProxy.PROXY_AUTHORIZATION;
+import static org.wildfly.channel.proxy.HttpProxy.asHttpHost;
+
+public class HttpUrlStreamHandler extends URLStreamHandler {
+
+    private HttpClient httpClient;
+
+    public HttpUrlStreamHandler() {
+        this(HttpClients.custom().useSystemProperties().build());
+    }
+
+    public HttpUrlStreamHandler(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public Settings getProxySettings(URL url) {
+        SettingsFactory settingsFactory = SettingsFactory.valueOf(url.getProtocol().toUpperCase());
+        return settingsFactory.createFromSystemProperties();
+    }
+
+
+    @Override
+    protected URLConnection openConnection(URL url, Proxy proxy) throws IOException {
+        throw new RuntimeException("Not Implemented yet");
```

Comment:
I removed it and let the default implementation:
```
throw new UnsupportedOperationException("Method not implemented.");
```

I think that should be good to implement it, to avoid any incompatibilities that can appear
Should be done is this PR?

Reply:
I'm not sure this is necessary. This change is not for an all-purpose application proxy. The execution is scoped to the Maven execution so we can live without it afaict.